### PR TITLE
feat(image): add can-preview & on-load prop and  add loading & errorbox slot

### LINKED ISSUE #1647: 否能增加一个Prop控制Image组件预览开关
### This function solves the problem (这个功能解决的问题)
1、当不想有预览时，目前好像只能使用Avatar或者原生img标签了
2、预览时的图片配置 max-height: 90vh; max-width: 90vw; 是否会更友好一些；当大图片时100vh 100vw太满了

### Expected API (期望的 API)
无

### PR DESCRIPTION
**Pull request by 2277419213**
close #1647

### CODE REVIEW

File: src/image/demos/enUS/index.demo-entry.md [hunk 1]
```diff
@@ -20,10 +22,19 @@ group
...
 | width | `string \| number` | `undefined` | Image width. |
 | on-error | `(e: Event) => void` | `undefined` | Callback executed when the image fails to load. |
+| on-load | `(e: Event) => void` | `undefined` | Callback executed after the image is loaded. |
+
+### Image Slots
+
+| Name     | Type | Description                                          |
+| -------- | ---- | ---------------------------------------------------- |
+| loading  | `()` | Excessive animation during image loading.            |
+| errorbox | `()` | A placeholder in the event of an image load failure. |
```
Comment (amadeus711) [inline src/image/demos/enUS/index.demo-entry.md:37]:
```suggestion
| error | `()` | A placeholder in the event of an image load failure. |
```

File: src/image/demos/zhCN/loading.demo.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# 加载图片
+
+仅仅只是演示加载效果，在图片加载的时候组件会自动显示加载的效果，`不推荐`直接修改 showLoadIng 属性，除非你很明确知道你在做什么！
```
Comment (amadeus711) [inline src/image/demos/zhCN/loading.demo.md:3]:
```suggestion
仅仅只是演示加载效果，在图片加载的时候组件会自动显示加载的效果，**不推荐** 直接修改 showLoadIng 属性，除非你很明确知道你在做什么！
```
`` 仅在 code 或 API 下使用，英文文档同

**Comment by 2277419213** [comment]
`n-switch` add `activeColor` & `inactiveColor` & `activeButtonColor` & `inactiveButtonColor` prop, closes [#1718](https://github.com/TuSimple/naive-ui/issues/1718)

File: src/image/demos/enUS/index.demo-entry.md [hunk 3]
```diff
@@ -20,10 +22,19 @@ group
 | img-props | `object` | `undefined` | The props of the img element inside the component. |
 | object-fit | `'fill' \| 'contain' \| 'cover' \| 'none' \| 'scale-down'` | `fill` | Object-fit type of the image in the container. |
 | preview-src | `string` | `undefined` | Source of preview image. |
+| can-preview | `boolean` | `true` | Whether clicking image preview is allowed. |
```
[the remark this answers is not in the document]
Comment (07akioni) [reply src/image/demos/enUS/index.demo-entry.md:25]: canBePreviewed

File: src/switch/demos/enUS/index.demo-entry.md [hunk 4]
```diff
@@ -29,6 +30,10 @@ shape
 | value | `boolean` | `undefined` | Value when being set manually. |
 | unchecked-value | `string \| boolean \| number` | `false` | Value of unchecked state. |
 | on-update:value | `(value: boolean) => void` | `undefined` | Callback when the component's value changes. |
+| activeColor | `string` | `undefined` | Background of checked state. |
+| inactiveColor | `string` | `undefined` | Background of unchecked state. |
+| activeButtonColor | `string` | `undefined` | Button background of checked state. |
+| inactiveButtonColor | `string` | `undefined` | Button background of unchecked state. |
```
Comment (07akioni) [inline src/switch/demos/enUS/index.demo-entry.md:36]: 文档里都用 kebab case

Comment (07akioni) [reply src/switch/demos/enUS/index.demo-entry.md:36]: 还有按照字母顺序

**Pull request merged**